float32 point_x1
float32 point_x2
float32 point_y1
float32 point_y2